pycobertura/filesystem.py

```diff
@@ -68,3 +69,52 @@ def open(self, filename):
             f.close()
         except KeyError:
             raise self.FileNotFound(filename)
+
+
+def get_root_path(repository_folder):
+    command = "git rev-parse --show-toplevel"
+    return get_output(command, working_folder=repository_folder).rstrip()
+
+
+class GitFileSystem(FileSystem):
+    def __init__(self, repo_folder, commit_id=None):
+        self.repository = repo_folder
+        self.commit_id = commit_id
+        self.repository_root = get_root_path(repo_folder)
+        self.prefix = self.repository.replace(self.repository_root, "").lstrip("/")
+
+    def real_filename(self, filename):
+        prefix = "{}/".format(self.prefix) if self.prefix else ""
+        return "{p.commit_id}:{prefix}{filename}".format(
+            prefix=prefix, p=self, filename=filename
+        )
```

Comment:
Can we instead do:

```suggestion
        return "{commit_id}:{prefix}{filename}".format(
            prefix=prefix, commit_id=self.commit_id, filename=filename
        )
```

... ? Would that work?

Reply:
Yeah, of course. When I first wrote this I was playing with format strings, and unfortunately this is still in the code.